kymatio/scattering1d/core/scattering1d.py

```diff
@@ -92,7 +96,8 @@ def scattering1d(x, pad, unpad, backend, J, psi1, psi2, phi, pad_left=0,
         # Convolution + downsampling
         j1 = psi1[n1]['j']
 
-        k1 = max(j1 - oversampling, 0)
+        sub1_adj = min(j1, log2_T) if average else j1
+        k1 = max(sub1_adj - oversampling, 0)
 
         assert psi1[n1]['xi'] < 0.5 / (2**k1)
```

Comment:
prod code doesn't use `assert` but rather raises specific errors

Reply:
`assert` is dropped in production code (compiled with "optimize" flag) for performance, yes

I think asserts make sense here for readability, if all tests pass there's little chance of deployment failure. Though it'd help to _explain_ why we assert, e.g. `# stability theorem, see [paper]`

Also unrelated to this PR

Reply:
> I think asserts make sense here for readability

`raise` adds even more readability